apis/v1alpha2/httproute_types.go

```diff
@@ -93,6 +93,13 @@ type HTTPRouteSpec struct {
 	// implementation must raise an 'Accepted' Condition with a status of
 	// `False` in the corresponding RouteParentStatus.
 	//
+	// If a Route (A) of type HTTPRoute or GRPCRoute is attached to a
+	// Listener and that listener already has another Route (B) of the other
+	// type attached and the intersection of the hostnames of A and B is
+	// non-empty, then the implementation must reject Route A. That is, the
+	// implementation must raise an 'Accepted' condition with a status of
+	// 'False' in the corresponding RouteParentStatus.
```

Comment:
Agree with Nick here, this is slightly inconsistent with our other conflict resolution guidance which generally relies on the creationTimestamp of the route and not when it was attached since creationTimestamp is more reliable for stateless controllers.

https://github.com/kubernetes-sigs/gateway-api/blob/ca079e6d90f2cb7521443dd5722a6c9fe345b8be/apis/v1beta1/httproute_types.go#L143-L159

Reply:
It seems like this could cause issues if someone tried to attach a route to a gateway with a route already attached. Is it possible for the controller to check for an `Accepted: true` condition at startup?

Reply:
Discussed in the community meeting. Going ahead and sticking in line with HTTPRoute/HTTPRoute conflict resolution for now but @youngnick is looking into solving this issue for the Gateway API in a more holistic manner.

Reply:
Refs https://github.com/kubernetes-sigs/gateway-api/issues/1211#issuecomment-1156710105 for prior discussion of whether checking status for conflict resolution is a viable path forward.